library/include/modules/EventManager.h

```diff
@@ -95,6 +96,12 @@ namespace DFHack {
             int32_t defendReport;
         };
 
+        struct ActionData {
+            int32_t unitId;
+            df::unit_action* action;
+            int32_t actionId;
```

Comment:
If `actionId` is a field of `action`, why pass the id as a separate field to the handler?

Reply:
Midunderstanding of the comment `//it has to keep the id of an item because the item itself may have been deallocated`, most likely; I didn't quite put together that this is for inventory_item in particular